docs/resources/templates/Design-Document-Template.md

```diff
@@ -0,0 +1,41 @@
+Design Title: 
+Authors: 
+Date: 
+
+## Overview
+This section explains the motivation of the work, providing the status QUO and the problems associated with it. This section also proposes the high level solution to the problem.
+
+## Goals
+This section explains the detailed goals and minimum requirements for the design. It should also include any non-goals of the design which would be expected from a reader. This also includes some high level functionality or performance gains.
+
+##	Use Cases
+This section explains the use cases which will be enabled by the implementation of this design. This section should be from the perspective of a user and how they would be able to interact with the feature described.
+
+## Design
+This section explains the high level changes proposed to achieve the design. Changes should be grouped by project components or in another logical way which makes the design easy to follow.
+If applicable, the design should be broken down into separate phases which each delivers incremental value. The first phase should achieve the minimum requirements outlined in the goals.
+If applicable, alternative approaches should be noted and evaluated for different sections of the design.
+
+## API Changes
+This is a section detailing any user facing or internal APIs added to components, including configuration changes. Any changes to user facing components are mentioned here (ie. WebUI, command line). If no changes are made, it should be stated in this section.
+Changes should be formatted in a reader friendly manner. A one-cell table should be used to contain the code blocks and monospace font should be used when writing code.
```

Comment:
Should the one-cell table only apply to google docs? Should we specify how it should be formatted for each type of format? (PDF, md, google doc...)

Reply:
Yes this line only applies to google docs. Removed.
I don't think we should specify how each type of file should be formatted, at least not in a template file in MD.